### PR TITLE
govCMS7 Automated Testing

### PR DESCRIPTION
**Pull request by pandaskii**
A rebased PR from https://github.com/govCMS/govCMS/pull/522

Reference review from 522:
https://github.com/govCMS/govCMS/pull/522#pullrequestreview-49370783

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,6 +84,65 @@ composer install --prefer-dist
...
+invoked from the parent directory of the build as follows:
+```
+build/bin/phing -f build/phing/build.xml test:phpunit
+```
+
+With the current configuration, the modules directory next to the docroot and
+its contents are searched, so tests will only be run for the custom govCMS
+modules. Files containing PHPunit tests should match the pattern '&ast;Test.php'.
+
+The iconomistTest.php may be used as a pattern for writing new tests for other
```
Comment (tobybellwood) [inline CONTRIBUTING.md]: Given we removed the iconomistTest - should we drop this para?

**Comment by pandaskii** [comment]
Testing is incompatible with PHP 7.*, and this is not a testing issue but whole govCMS7 platform.

**Comment by pandaskii** [comment]
Remove external module tdd7 from this PR (it is risky that any module is neither on Drupal nor custom folder)

**Comment by pandaskii** [comment]
snip20171204_2

Testing failed on PHP7.1 but if PHP5.6.* can pass the test, it is ok to merge this PR

**Comment by pandaskii** [comment]
image

Needs more tweaking on features

**Comment by pandaskii** [comment]
Selenium is getting many troubles with different drivers.

My local tests of this PR returned many failures but I am with Firefox v45 in a Drupal VM

83 scenarios (60 passed, 23 failed)
1137 steps (914 passed, 23 failed, 200 skipped)
34m4.86s (105.11Mb)

screen shot 2017-12-05 at 17 37 05

Let us see if govCMS Bot build failed or not, and blocked until the tests can be run in a stable environment

**Comment by tobybellwood** [comment]
I'll get #604 merged once it's passed testing, and then we can rerun the build bot.

**Comment by tobybellwood** [comment]
Test this

**Pull request merged**